### PR TITLE
Benchamark test for QoS

### PR DESCRIPTION
**Pull request by hujw77**
- benchmark test
- change the lane params
- change filter `call` to `staticcall` at inlane

### CODE REVIEW

File: contracts/bridge/contracts/common/message/InboundLane.sol [hunk 1]
```diff
@@ -304,11 +304,11 @@ contract InboundLane is MessageVerifier, SourceChain, TargetChain {
         }
     }
 
-    function filter(address target, bytes memory encoded) internal returns (bool canCall) {
+    function filter(address target, bytes memory encoded) internal view returns (bool canCall) {
         /**
          * @notice The app layer must implement the interface `ICrossChainFilter`
          */
-        (bool ok, bytes memory result) = target.call{value: 0, gas: GAS_BUFFER}(encoded);
+        (bool ok, bytes memory result) = target.staticcall{gas: GAS_BUFFER}(encoded);
```
Comment (HackFisher) [inline contracts/bridge/contracts/common/message/InboundLane.sol:311]: Would like to learn the difference here?

And does this ensure the implementation of `ICrossChainFilter` interface?

Comment (HackFisher) [reply contracts/bridge/contracts/common/message/InboundLane.sol:311]: > And does this ensure the implementation of ICrossChainFilter interface?

I see, the encoded is only called from dispatch, and ICrossChainFilter's selector is encoded there.

Comment (hujw77) [reply contracts/bridge/contracts/common/message/InboundLane.sol:311]: * `STATICCALL` is used when view functions are called, which enforces the state to stay unmodified as part of the EVM execution, and the `function crossChainFilter(uint32 bridgedChainPosition, uint32 bridgedLanePosition, address sourceAccount, bytes calldata payload) external view returns (bool)` interface is view.
* `Call` do a message call like a transaction.

Comment (HackFisher) [reply contracts/bridge/contracts/common/message/InboundLane.sol:311]: Good move:

https://medium.com/blockchannel/state-specifiers-and-staticcall-d50d5b2e4920

**Review by HackFisher — APPROVED** [review]

**Pull request merged**